coverage/ctracer/util.h

```diff
@@ -5,26 +5,29 @@
 #define _COVERAGE_UTIL_H
 
 #include <Python.h>
+#include "frameobject.h"          // PyFrameObject
 
 /* Compile-time debugging helpers */
 #undef WHAT_LOG         /* Define to log the WHAT params in the trace function. */
 #undef TRACE_LOG        /* Define to log our bookkeeping. */
 #undef COLLECT_STATS    /* Collect counters: stats are printed when tracer is stopped. */
 #undef DO_NOTHING       /* Define this to make the tracer do nothing. */
 
-#if PY_VERSION_HEX >= 0x030B00A0
-// 3.11 moved f_lasti into an internal structure. This is totally the wrong way
-// to make this work, but it's all I've got until https://bugs.python.org/issue40421
-// is resolved.
-#include <internal/pycore_frame.h>
-#define MyFrame_lasti(f)    ((f)->f_frame->f_lasti * 2)
-#elif PY_VERSION_HEX >= 0x030A00A7
-// The f_lasti field changed meaning in 3.10.0a7. It had been bytes, but
-// now is instructions, so we need to adjust it to use it as a byte index.
-#define MyFrame_lasti(f)    ((f)->f_lasti * 2)
-#else
-#define MyFrame_lasti(f)    ((f)->f_lasti)
-#endif
+static inline Py_ssize_t MyFrame_lasti(PyFrameObject *frame)
+{
+    PyObject *obj = PyObject_GetAttrString((PyObject*)frame, "f_lasti");
```

Comment:
I guess I shouldn't be concerned about the overhead of this function?

Reply:
I don't know how to measure that. If you measure that the overhead is too large, I can consider adding a getter function to Python 3.11.

Does coverage performance overhead matter in general? Usually, I don't care much about performance of debugging or profiler tools.

Reply:
People run their test suites under coverage, and test suites are always too slow.  "debugging and profiling" is a more occasional activity.  I'll take a look at it.

Reply:
Right, I get it. But I would prefer to not add getter functions just for coverage if the performance overhead of this PR is "acceptable". I would prefer to only consider new functions to the Python C API if it's "needed". So it would be nice if you could provide a benchmark (and benchmark results) so measure the performance impact of this PR. Sorry, I don't know how to do that.